### DISCUSSION TITLE
Laravel 11: Can you get exceptions in middleware？

### DISCUSSION THREAD

**Discussion by Mr-wangwei** [laravel/framework#52576] [Q&A]
Can you get exceptions in middleware?
How to log requests log and add exceptions?

**Comment by ozdemirrulass** [comment]
I don't know why you want to do this in the middleware but yes you can.

```php
public function handle(Request $request, Closure $next)
    {
        try {
            Log::info('Incoming Request', [
                'url' => $request->fullUrl(),
                'method' => $request->method(),
                'headers' => $request->headers->all(),
                'body' => $request->all()
            ]);

            $response = $next($request);

            Log::info('Response Status', ['status' => $response->getStatusCode()]);

            return $response;
        } catch (Throwable $e) {
            Log::error('Exception Occurred', [
                'message' => $e->getMessage(),
                'stack' => $e->getTraceAsString(),
            ]);

            throw $e;
        }
    }
```

**Reply by Mr-wangwei** [reply]
Thanks for the answer.
I've tried this and it doesn't catch the exception.
I want to write the request log (containing the exceptions generated in the request) to Mysql or Elasticsearch, is there any good way to do it?

**Reply by ozdemirrulass** [reply]
If an exception is thrown after the middleware has finished executing (e.g., in the controller, service layer, or after a database transaction), it won't be caught in the middleware. Laravel's default exception handling flow is designed to catch and handle exceptions globally, meaning middleware typically operates before the request reaches the controller or after the controller returns a response. This is why I said I don't know why you want to do this in the middleware.
